File: 78-subsets/subsets.java
class Solution {
    List<List<Integer>> res = new LinkedList<>();
    List<Integer> path = new LinkedList<>();
    public List<List<Integer>> subsets(int[] nums) {
        helper(nums, 0);
        return res;
    }

    public void helper(int[] nums, int index) {
        res.add(new LinkedList<>(path));
        for (int i = index; i < nums.length; i++) {
            path.add(nums[i]);
            helper(nums, i+ 1);
            path.removeLast();
        }
    }
}

File: 78-subsets/subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        self.res = []
        self.helper(nums, 0, [])
        return self.res
    def helper(self, nums, idx, temp):
        # if idx >= len(nums):
        #     return
        self.res.append(temp[:])
        for i in range(idx, len(nums)):
            temp.append(nums[i])
            self.helper(nums, i + 1, temp)
            temp.pop()
        